### PR TITLE
hack/doc: Autogen CLI Documentation

### PR DESCRIPTION
**Pull request by theishshah**
**Description of the change:**

Adds `make doc` command to generate CLI documentation

**Motivation for the change:**

Reduce developer friction when making changes to the CLI

Includes sample generated documentation

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -56,6 +56,9 @@ tidy: ## Update dependencies
 clean: ## Clean up the build artifacts
 	$(Q)rm -rf build
 
+doc: ## Generate CLI Documentation
```
Comment (asmacdo) [inline Makefile]: [Optional] Could this be `cli-doc`?

Comment (theishshah) [reply Makefile]: agree, done.

File: doc/cli/operator-sdk_add_api.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+By default, this command runs Kubernetes deepcopy and OpenAPI V3 generators on
+tagged types in all paths under pkg/apis. Go code is generated under
+pkg/apis/<group>/<version>/zz_generated.{deepcopy,openapi}.go. CRD's are
+generated, or updated if they exist for a particular group + version + kind,
+under deploy/crds/<full group>_<resource>_crd.yaml; OpenAPI V3 validation YAML
+is generated as a 'validation' object. Generation can be disabled with the
+--skip-generation flag.
+
+Example:
```
[the remark this answers is not in the document]
Comment (theishshah) [reply doc/cli/operator-sdk_add_api.md]: should be fixed now

File: hack/doc/gen_cli_doc.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+go run ./hack/doc/gen-cli-doc.go
```
Comment (estroz) [inline hack/doc/gen_cli_doc.sh]: Newline here

File: hack/doc/gen-cli-doc.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+package main
+
+import (
+	"os"
+
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/cli"
+
+	"github.com/spf13/cobra/doc"
+
```
Comment (estroz) [inline hack/doc/gen-cli-doc.go]:
```suggestion
```

File: hack/doc/gen-cli-doc.go [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+	log "github.com/sirupsen/logrus"
+)
+
+func main() {
+	root := cli.GetCLIRoot()
+	currentDir, err := os.Getwd()
+	if err != nil {
+		log.Fatalf("Failed to get current directory, (%v)", err)
```
Comment (estroz) [inline hack/doc/gen-cli-doc.go]:
```suggestion
		log.Fatalf("Failed to get current directory: %v", err)
```

File: hack/doc/gen-cli-doc.go [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+	}
+
+	err = doc.GenMarkdownTree(root, currentDir+"/doc/cli")
+	if err != nil {
+		log.Fatalf("Failed to generate documenation, (%v)", err)
```
Comment (estroz) [inline hack/doc/gen-cli-doc.go]:
```suggestion
		log.Fatalf("Failed to generate documenation: %v", err)
```

**Pull request merged**